thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/route.js

```diff
@@ -364,64 +362,17 @@ export default Route.extend({
       });
     });
 
-    // Fetch all anomalies we have Ids for. Enhance the data and populate power-select filter options.
-    // TODO: look into possibility of bundling calls or using async/await: https://github.com/linkedin/pinot/pull/2468
-    if (notCreateError) {
-      fetchCombinedAnomalies(anomalyIds)
-        .then((rawAnomalyData) => {
-          rawAnomalies = rawAnomalyData;
-          return fetchSeverityScores(anomalyIds);
-        })
-        .then((severityScores) => {
-          anomalyData = enhanceAnomalies(rawAnomalies, severityScores);
-          resolutionOptions.push(...new Set(anomalyData.map(record => record.anomalyFeedback)));
-          dimensionOptions.push(...new Set(anomalyData.map(anomaly => anomaly.dimensionString)));
-          controller.setProperties({
-            anomaliesLoaded: true,
-            anomalyData,
-            resolutionOptions,
-            dimensionOptions
-          });
-          return this.fetchCombinedAnomalyChangeData(anomalyData);
-        })
-        // Load and display rest of options once data is loaded ('2week', 'Last Week')
-        .then((wowData) => {
-          anomalyData.forEach((anomaly) => {
-            anomaly.wowData = wowData[anomaly.anomalyId] || {};
-          });
-          controller.setProperties({
-            anomalyData,
-            baselineOptions: [baselineOptions[0], ...newWowList]
-          });
-          return fetch(metricDataUrl).then(checkStatus);
-        })
-        // Fetch and load graph metric data. Stop spinner/loader
-        .then((metricData) => {
-          Object.assign(metricData, { color: metricDataColor });
-          controller.setProperties({
-            metricData,
-            alertDimension,
-            topDimensions: [],
-            isMetricDataLoading: false
-          });
-          // If alert has dimensions set, load them into graph once replay is done.
-          if (exploreDimensions && !isReplayPending) {
-            controller.set('topDimensions', getTopDimensions(metricData, dimensionCount));
-          }
-        })
-        .catch((errors) => {
-          controller.setProperties({
-            isMetricDataInvalid: true,
-            isMetricDataLoading: false,
-            graphMessageText: 'Error loading metric data'
-          });
-        });
+    // Begin loading anomaly and graph data as concurrency tasks
+    if (jobId !== -1) {
```

Comment:
did you know that can use the bitwise not operator `~-1 == 0` so the check could be rewritten as `if (~jobId)`
This is just an fyi, some people might think this is a very obscure way to write things 

Documentation: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_Operators#Bitwise_NOT

Reply:
I'll look into this.